config/clients/dotnet/template/Telemetry/Attributes.cs.mustache

```diff
@@ -0,0 +1,227 @@
+
+    // The client ID associated with the request, if any
+    private const string AttributeRequestClientId = "fga-client.request.client_id";
+
+    // Attributes (tags) associated with the response //
+
+    // The authorization model ID that the FGA server used
+    private const string AttributeResponseModelId = "fga-client.response.model_id";
+
+    // Attributes (tags) associated with specific actions //
+
+    // The user that is associated with the action of the request for check and list objects
+    private const string AttributeFgaRequestUser = "fga-client.user";
+
+    // OTEL Semantic Attributes (tags) //
+
+    // The total request time for FGA requests
+    private const string AttributeHttpClientRequestDuration = "http.client.request.duration";
+
+    // The amount of time the FGA server took to internally process nd evaluate the request
+    private const string AttributeHttpServerRequestDuration = "http.server.request.duration";
+
+    // The HTTP method for the request
+    private const string AttributeHttpMethod = "http.request.method";
+
+    // The status code of the response
+    private const string AttributeHttpStatus = "http.response.status_code";
+
+    // Host identifier of the origin the request was sent to
+    private const string AttributeHttpHost = "http.host";
+
+    // HTTP Scheme of the request (`http`/`https`)
+    private const string AttributeHttpScheme = "url.scheme";
+
+    // Full URL of the request
+    private const string AttributeHttpUrl = "url.full";
+
+    // User Agent used in the query
+    private const string AttributeHttpUserAgent = "user_agent.original";
+
+    // The number of retries attempted (starting from 1 for the original request)
+    private const string AttributeRequestRetryCount = "http.request.resend_count";
+
+    private static string? GetHeaderValueIfValid(HttpResponseHeaders headers, string headerName) {
+        if (headers.Contains(headerName) && headers.GetValues(headerName).Any()) {
+            return headers.GetValues(headerName).First();
+        }
+
+        return null;
+    }
+
+    /**
+     * Builds an object of attributes that can be used to report alongside an OpenTelemetry metric event.
+     *
+     * @param response - The Axios response object, used to add data like HTTP status, host, method, and headers.
+     * @param credentials - The credentials object, used to add data like the ClientID when using ClientCredentials.
+     * @param methodAttributes - Extra attributes that the method (i.e. check, listObjects) wishes to have included. Any custom attributes should use the common names.
+     * @returns {Attributes}
+     */
+    public static KeyValuePair<string, object>[] buildAttributesForResponse(string apiName,
+        HttpResponseMessage response, RequestBuilder requestBuilder, Credentials? credentials,
+        Stopwatch requestDuration, int retryCount) {
+        var attributes = new List<KeyValuePair<string, object>>();
```

Comment:
Use & return a `TagList` rather than a list of KVPs. It's a type created for this specific usage. Metric recording types support it as an overload.